test/parallel/test_tensorflow.py

```diff
@@ -117,19 +117,22 @@ def test_horovod_allreduce_cpu(self):
         """Test on CPU that the allreduce correctly sums 1D, 2D, 3D tensors."""
         hvd.init()
         size = hvd.size()
-        dtypes = self.filter_supported_types([tf.int32, tf.int64, tf.float16, tf.float32, tf.float64])
+        dtypes = self.filter_supported_types([tf.uint8, tf.int8, tf.int32, tf.int64, tf.float16, tf.float32, tf.float64])
         dims = [1, 2, 3]
         for dtype, dim in itertools.product(dtypes, dims):
             with tf.device("/cpu:0"):
-                tensor = self.random_uniform(
-                    [17] * dim, -100, 100, dtype=dtype)
+                # prevent underflows/overflows in uint8, int8
+                tensor = self.random_uniform([17] * dim, 0, 10)
```

Comment:
I'd suggest to leave range at [-100,100] in general and add a special case for `uint8` if necessary (like taking the absolute value and casting). 

Removing any negative tensor values from the test coverage does not sound wise and I'd also prefer not to change the range with this PR where it's not necessary.

(comment applies to all tests adapted for `(u)int8`)

Reply:
sure, will change it.